test/async_cache_test.dart

```diff
@@ -158,4 +158,20 @@ void main() {
     }));
     expect(cache.fetchStream(call).toList(), completion(['1', '2', '3']));
   });
+
+  test('should invalidate even if the future throws an exception', () async {
+    cache = AsyncCache.ephemeral();
+
+    Future<String> throwingCall() async => throw Exception();
+    try {
+      await cache.fetch(throwingCall);
+    } catch (exception) {
+      expect(exception, isException);
+    }
```

Comment:
Use:
```dart
 await expectLater(cache.fetch(throwingCall), throwsA(isException));
```
so that you check that it actually throws. The code above will succeed even if it *doesn't* throw.

Reply:
I've changed the code to the suggested one, it works well.